### DISCUSSION TITLE
digitalWrite(LED_BUILTIN,0) has inverted logic

### DISCUSSION THREAD

**Discussion by FabioM3** [stm32duino/Arduino_Core_STM32#2580] [Q&A] [answered]
**Describe the bug**
The logic on "digitalWrite(LED_BUILTIN, 0)" is inverted, when I use an external led it works, but for the LED_BUILTIN it turns OFF the led with 1 and ON with 0. 

**To Reproduce**

`#define botaoLigaDesliga A3
short ligado = 0;
unsigned long tempoPassadoBotao = 0;

void setup() {
  pinMode(botaoLigaDesliga, INPUT_PULLDOWN);
  pinMode(LED_BUILTIN, OUTPUT);
 
  digitalWrite(LED_BUILTIN, ligado);
}

void loop() {
  if (digitalRead(botaoLigaDesliga)) {
    if (millis() - tempoPassadoBotao >= 2000) {
      if (ligado) {
        ligado = 0;
      } else {
        ligado = 1;
      }

      digitalWrite(LED_BUILTIN, ligado);
      tempoPassadoBotao = millis();
      Serial.print("Botão apertado - ");
      Serial.println(ligado);
    }
  } 
}`

1. Plug.
2. Press BTN on A3 to change the state of the led.
3. See on serial the 0 or 1 and compare with the led.

**Expected behavior**
The led needs to be OFF with 0 and on with 1.

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Arduino IDE version: [e.g. 2.3]
 - STM32 core version: [e.g. 2.9]
 
![image](https://github.com/user-attachments/assets/cb80f34a-a4ed-4af8-94c1-d7972c3211f2)

**Comment by fpistm** [comment, marked as answer]
Hi @FabioM3 
This is not an issue. This is only linked to the hardware, the Active state is inverted
![image](https://github.com/user-attachments/assets/9b78e5a7-2c2c-4448-8637-c1f3e0477032)

Logical level is:
HIGH --> LED OFF
LOW --> LED ON

**Comment by FabioM3** [comment]
Oh, I thought it was something hardware related, but I haven't found anything related to this in my searches.
At least now people have this post as a reference to this bad design.
Thanks.
